eventuate-adapter-vertx/src/main/scala/com/rbmhtechnology/eventuate/adapter/vertx/VertxReader.scala

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2015 - 2016 Red Bull Media House GmbH <http://www.redbullmediahouse.com> - all rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.rbmhtechnology.eventuate.adapter.vertx
+
+import com.rbmhtechnology.eventuate.EventsourcedWriter
+import com.rbmhtechnology.eventuate.adapter.vertx.api.{ StorageProvider, VertxEndpointRouter }
+import io.vertx.core.eventbus.impl.MessageImpl
+import io.vertx.core.eventbus.{ DeliveryOptions, EventBus, Message }
+import io.vertx.core.{ AsyncResult, Handler, Vertx, Future => VertxFuture }
+
+import scala.collection.immutable.Seq
+import scala.concurrent.duration.{ FiniteDuration, _ }
+import scala.concurrent.{ ExecutionContext, Future, Promise }
+
+case class EventEnvelope(address: String, event: Any)
+
+trait MessageProducer {
+  import VertxHandlerConverters._
+
+  protected def eventBus(): EventBus =
+    vertx.eventBus()
+
+  def vertx: Vertx
+
+  protected def produce[A](address: String, msg: Any, deliveryOptions: DeliveryOptions, handler: Handler[AsyncResult[Message[A]]]): Unit
```

Comment:
In this case I would say yes, because from a  sender point-of-view a publish can be seen as a send with an immediate response.
This enables generic handling of message-sending in the `VertxReader`